manifests/config/global.pp

```diff
@@ -18,15 +19,13 @@
   $flattendata = $newtype.keys.reduce({}) |$memo, $key| { $memo + {$key => $newtype[$key]["value"]} }
 
   unless empty($flattendata) {
-    concat::fragment { 'rsyslog::component::global_config::newtype':
-      order   => $::rsyslog::global_config_priority,
-      target  => "${::rsyslog::confdir}/${::rsyslog::target_file}",
-      content => epp('rsyslog/global_config.epp',
-      {
-        'type'   => 'rainerscript',
-        'config' => $flattendata,
-      }),
+    rsyslog::component::global_config { 'rainerscript':
+      *      => {
```

Comment:
What youre referring to is defaults - that has nothing to do with '*' (although you can use it in the same way) - you're thinking of the special default keyword

```
resource {
  default:
     ensure => present,
     some  => thing,

  'title1':
     other => thing,
} 
```